### PR TITLE
Add daily cron CI job that runs slow tests

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

This commit adds a new custom CI job to azure pipelines that will run a
full unit test job include the slow tests. Right now we're not
exercising the slow tests as part of CI this commit closes this gap so
that we verify that the tests at least run.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -764,3 +764,88 @@ stages:
...
         - bash: python -m unittest discover -v test/ipynb
           displayName: 'Run image test'
+
+schedules:
+- cron: "0 6 * * *"
+  displayName: Nightly slow build
+  branches:
+    include:
+      - master
+  jobs:
```
Comment (mtreinish) [inline azure-pipelines.yml:774]: Azure doesn't like this, it failed validation on this line. I'll have to read the docs some more about the syntax for scheduled jobs.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -764,3 +764,88 @@ stages:
...
+          inputs:
+            contents: '**/*.png'
+            targetFolder: $(Build.ArtifactStagingDirectory)
+        - task: PublishBuildArtifacts@1
+          condition: failed()
+          displayName: 'Publish images'
+          inputs:
+            pathtoPublish: '$(Build.ArtifactStagingDirectory)'
+            artifactName: 'drop_linux'
+        - bash: |
```
Comment (kdk) [inline azure-pipelines.yml:839]: Will this/can this generate any kind of push notification on failure?

Comment (mtreinish) [reply azure-pipelines.yml:898]: Not by default, azure's built in notifications are only for users with a linked azure devops account (which I think we both have), but they only offer a limit number of accounts to free organizations. So it's not a general solution (although you probably could personally subscribe to notifications for this https://docs.microsoft.com/en-us/azure/devops/notifications/about-notifications?view=azure-devops). I think the only way for this to work is to leverage the `tools/report_ci_failure.py` and run it in the job (and make sure we set the secret with the token for the @QacoBot account).

**Comment by mtreinish** [comment]
This is superseded by #8811

**Pull request closed without merging**